src/components/forms/Form.js

```diff
@@ -182,42 +165,58 @@ class Form extends Component {
 				}
 			)
 		).isRequired,
-		initialValues: PropTypes.arrayOf(
-			PropTypes.shape(
-				{
-					camelCaseName: PropTypes.string.isRequired,
-					value: PropTypes.any
-				}
-			)
-		),
-		title: PropTypes.string.isRequired,
-		onSubmit: PropTypes.func.isRequired,
+		initialValues: PropTypes.object,
+		title: PropTypes.string,
+		onSubmit: PropTypes.func,
+		onChange: PropTypes.func,
 		visible: PropTypes.bool.isRequired,
-		changeVisibility: PropTypes.func.isRequired,
-		onCancel: PropTypes.func,
+		changeVisibility: PropTypes.func,
 		submitButtonName: PropTypes.string
 	}
 
-	changeState(name, val = "") {
-		this.setState({
-			[name]: val
+	componentDidUpdate(prevProps) {
+		if (!_.isEqual(prevProps.initialValues, this.props.initialValues)) this.setState(this.props.initialValues);
```

Comment:
The form can now be controlled by the parent like most other components. There are gonna be times where the form's values prop is updated and the states in the form should be updated as well. Otherwise every parent would need to use a ref if they wanted to to update the states in the form which I think is less recommended. The form inside MultiElement is an example.